cmd/oras/root/push.go

```diff
@@ -108,6 +108,21 @@ Example - Push file "hi.txt" into an OCI image layout folder 'layout-dir' with t
 			if err := option.Parse(cmd, &opts); err != nil {
 				return err
 			}
+
+			if opts.manifestConfigRef != "" && opts.artifactType == "" {
+				// Here packs an artifact that is not compliant to OCI artifact guidance.
+				// Reference: https://github.com/opencontainers/image-spec/blob/e7f7c0ca69b21688c3cea7c87a04e4503e6099e2/manifest.md?plain=1#L170
+				switch opts.Flag {
+				case option.ImageSpecAuto:
+					opts.PackVersion = oras.PackManifestVersion1_0
```

Comment:
```suggestion
				switch opts.Flag {
				case option.ImageSpecAuto:
					// switch to v1.0 manifest since artifact type is suggested by OCI v1.1 artifact guidance but not presented 
					// see https://github.com/opencontainers/image-spec/blob/e7f7c0ca69b21688c3cea7c87a04e4503e6099e2/manifest.md?plain=1#L170
					opts.PackVersion = oras.PackManifestVersion1_0
```